3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,320 @@
+
+		// enforce freedom mode for videopress
+		add_filter( 'videopress_shortcode_options', array( 'Jetpack_AMP_Support', 'videopress_enable_freedom_mode' ) );
+
+		// include Jetpack og tags when rendering native AMP head
+		add_action( 'amp_post_template_head', array( 'Jetpack_AMP_Support', 'amp_post_jetpack_og_tags' ) );
+
+		// Post rendering changes for legacy AMP
+		add_action( 'pre_amp_render_post', array( 'Jetpack_AMP_Support', 'amp_disable_the_content_filters' ) );
+
+		// Add post template metadata for legacy AMP
+		add_filter( 'amp_post_template_metadata', array( 'Jetpack_AMP_Support', 'amp_post_template_metadata' ), 10, 2 );
+	}
+
+	static function admin_init() {
+		// disable Likes metabox for post editor if AMP canonical disabled
+		add_filter( 'post_flair_disable',  array( 'Jetpack_AMP_Support', 'is_amp_canonical' ), 99 );
+	}
+
+	static function init_filter_jetpack_widgets() {
+		if ( ! self::is_amp_request() ) {
+			return;
+		}
+
+		// widgets
+		add_filter( 'jetpack_widgets_to_include', array( 'Jetpack_AMP_Support', 'filter_available_widgets' ) );
+	}
+
+	static function is_amp_canonical() {
+		return function_exists( 'amp_is_canonical' ) && amp_is_canonical();
+	}
+
+	static function is_amp_request() {
+		// can't use is_amp_endpoint() since it's not ready early enough in init.
+		// is_amp_endpoint() implementation calls is_feed, which bails with a notice if plugins_loaded isn't finished
+		// "Conditional query tags do not work before the query is run"
+		$is_amp_request = ! is_admin() // this is necessary so that modules can still be enabled/disabled/configured as per normal via Jetpack admin
+			&&
+				function_exists( 'amp_is_canonical' ) // this is really just testing if the plugin exists
+			&&
+				( amp_is_canonical() || isset( $_GET[ amp_get_slug() ] ) );
+
+		/**
+		 * Returns true if the current request should return valid AMP content.
+		 *
+		 * @since 6.2.0
+		 *
+		 * @param boolean $is_amp_request Is this request supposed to return valid AMP content?
+		 */
+		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
+	}
+
+	static function filter_available_widgets( $widgets ) {
+		if ( self::is_amp_request() ) {
+			$widgets = array_filter( $widgets, array( 'Jetpack_AMP_Support', 'is_supported_widget' ) );
+		}
+
+		return $widgets;
+	}
+
+	static function is_supported_widget( $widget_path ) {
+		return substr($widget_path, -14) !== '/milestone.php';
+	}
+
+	static function amp_disable_the_content_filters() {
+		if ( defined( 'WPCOM') && WPCOM ) {
+			add_filter( 'videopress_show_2015_player', '__return_true' );
+			add_filter( 'protected_embeds_use_form_post', '__return_false' );
+			remove_filter( 'the_title', 'widont' );
+		}
+
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'filter' ), 11 );
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'maybe_create_links' ), 100 );
+	}
+
+	/**
+	 * Add publisher and image metadata.
+	 *
+	 * @since 0.3
```

Comment:
Should these methods be `6.2.0` instead? I imagine that they're brought over from `amp-wp` and were probably added in that version there.

Perhaps it could be something like:

```
@since 0.3 In amp-wp plugin
@since 6.2.0 Ported to Jetpack
```

I'm not sure. It's also pretty minor.

Reply:
Fixed in new PR